tests/tracking/test_rest_tracking.py

```diff
@@ -288,11 +288,18 @@ def test_log_metrics_params_tags(mlflow_client, backend_store_uri):
     created_run = mlflow_client.create_run(experiment_id)
     run_id = created_run.info.run_id
     mlflow_client.log_metric(run_id, key='metric', value=123.456, timestamp=789, step=2)
+    mlflow_client.log_metric(run_id, key='nan_metric', value=float("nan"))
+    mlflow_client.log_metric(run_id, key='inf_metric', value=float("inf"))
+    mlflow_client.log_metric(run_id, key='-inf_metric', value=-float("inf"))
     mlflow_client.log_metric(run_id, key='stepless-metric', value=987.654, timestamp=321)
     mlflow_client.log_param(run_id, 'param', 'value')
     mlflow_client.set_tag(run_id, 'taggity', 'do-dah')
     run = mlflow_client.get_run(run_id)
     assert run.data.metrics.get('metric') == 123.456
+    import math
+    assert math.isnan(run.data.metrics.get('nan_metric'))
```

Comment:
Extra credit but could we add a test verifying that @max-allen-db's panda converter handles the NaNs correctly? https://github.com/mlflow/mlflow/blob/master/tests/tracking/test_fluent.py#L373

Reply:
I'd rather not add anything more to this PR and get this merged in time for the release. Can we add it as a followup?